internal/provider/validators.go

```diff
@@ -178,6 +179,42 @@ func validateConsulClusterSize(v interface{}, path cty.Path) diag.Diagnostics {
 	return diagnostics
 }
 
+func validateConsulClusterCIDR(v interface{}, path cty.Path) diag.Diagnostics {
+	var diagnostics diag.Diagnostics
+
+	addr := v.(string)
+	ip, err := netip.ParsePrefix(addr)
+	isIPV4 := ip.Addr().Is4()
+
+	if err != nil || !isIPV4 {
```

Comment:
Probably worth checking the validity of the prefix, too:
```suggestion
	if err != nil || !ip.IsValid() || !ip.Addr().Is4() {
```

Reply:
yep thanks!